### PR TITLE
feat(connect): add Solana support to Connect

### PR DESCRIPTION
**Pull request by gabrielKerekes**
## Description

Extracted Solana Connect changes into a separate PR as requested. This is based on https://github.com/trezor/trezor-suite/pull/9921 so only the `feat(connect): ...` commits (starting with 174e52e9bcc3b6141006090ef550b1f97d8f1f26) should be reviewed here.

### CODE REVIEW

File: packages/connect/e2e/__fixtures__/solanaGetAddress.ts
Comment (mroz22) [inline packages/connect/e2e/__fixtures__/solanaGetAddress.ts:10]: do you have tests fixtures in firmware repo? if yes, preferred way would be to import them from submodules folder probably to avoid duplication

Comment (gabrielKerekes) [reply packages/connect/e2e/__fixtures__/solanaGetAddress.ts:10]: Yes there are tests in the FW repo and yes they are also the same. I'm not sure what the steps would be to be able to import the tests here. The FW PR has still not been merged and so the `trezor-common` repo has not been updated as well - I don't know what the process is to update it.

To be honest I'd prefer if we didn't block merging this PR on this detail, we can refactor the tests once `trezor-common` has been updated.

Probably only the `solanaGetAddress` and `solanaGetPublicKey` tests could be reused from FW - we haven't really gotten feedback yet on how to handle the sign tx test on FW but they most likely won't be usable in Connect.

And I just noticed that the `solanaSignTransaction` tests are missing the `path` parameter - I will fix this.

Comment (gabrielKerekes) [reply packages/connect/e2e/__fixtures__/solanaGetAddress.ts:10]: > And I just noticed that the solanaSignTransaction tests are missing the path parameter - I will fix this.

Fixed in https://github.com/trezor/trezor-suite/pull/9958/commits/2f51430f6a270797021b5d7c271b49fccfa3d598.

**Comment by mroz22** [comment]
please rebase on develop now 🙏

**Comment by gabrielKerekes** [comment]
> please rebase on develop now 🙏

Done.

File: packages/connect/e2e/__fixtures__/solanaGetAddress.ts
Comment (mroz22) [reply packages/connect/e2e/__fixtures__/solanaGetAddress.ts:10]: ok

File: packages/connect-common/files/coins.json
Comment (mroz22) [inline packages/connect-common/files/coins.json:2215]: note: need to check this. 2.6.1 is probably not right but I am not sure if these values are used at the moment

File: docs/packages/connect/methods/solanaGetAddress.md
Comment (mroz22) [inline docs/packages/connect/methods/solanaGetAddress.md:2]: Since we are going to merge this before official firmware lands master branch, I suggest adding something like 

```
> :warning: **BETA**: This api is unstable and may change!

```

to all readmes and console.warn into every newly added connect methods init member.
just to be sure that people don't start using it right away and we don't have to wrap our heads around backwards compatibility in the future.

File: packages/connect/e2e/__fixtures__/getFeatures.ts
Comment (mroz22) [inline packages/connect/e2e/__fixtures__/getFeatures.ts]: maybe lets remove changes to this file. once firmware is merged and becomes available in CI, we will adjust. 

other newly added test files may stay, although they are not executed at the moment (it would require to add them to test.yml file). We don't want to do it until firmware is avaiable in master.

File: packages/connect/src/data/config.ts
Comment (mroz22) [inline packages/connect/src/data/config.ts]: I guess, there should be 3 records? T2B1 is imho missing here

File: packages/connect/src/data/defaultFeeLevels.ts
Comment (mroz22) [inline packages/connect/src/data/defaultFeeLevels.ts:99]: I have zero knowledge about fees in solana. this means there is only static fee? always 5000?

**Review by mroz22 — COMMENTED** [review]
it looks ok. just couple of comments, please take a look. just reading the code, I am almost ready to merge this. 

@marekrjpolak is going to examine it as well, so lets wait for him.

File: packages/connect/src/data/defaultFeeLevels.ts
Comment (gabrielKerekes) [reply packages/connect/src/data/defaultFeeLevels.ts:99]: Hm, it's actually 5000*`number_of_signatures`. Solana tx size is limited to 1232B, one signature + public key is 96B so the limit is 5000*(1232/96) -> 12. I'm not sure how this value is used in Suite so I'm not sure if we should change it to 5000*12. @memoriesadrift could you please take a look? 🙏

File: docs/packages/connect/methods/solanaGetAddress.md
Comment (gabrielKerekes) [reply docs/packages/connect/methods/solanaGetAddress.md:2]: Good point. Added in https://github.com/trezor/trezor-suite/pull/9958/commits/5b886ffb3d989c4b945426d52e46f4a472ae0c56.

File: packages/connect/e2e/__fixtures__/getFeatures.ts
Comment (gabrielKerekes) [reply packages/connect/e2e/__fixtures__/getFeatures.ts]: Removed -> https://github.com/trezor/trezor-suite/pull/9958/commits/5b886ffb3d989c4b945426d52e46f4a472ae0c56#diff-54c4e346122390d29d5d873858065141d2e563d80925cf0862505c251b04ca7dL80.

File: packages/connect/src/data/config.ts
Comment (gabrielKerekes) [reply packages/connect/src/data/config.ts]: Updated. I also removed the `comment` prop which was actually copy/pasted and false I think. -> https://github.com/trezor/trezor-suite/pull/9958/commits/5b886ffb3d989c4b945426d52e46f4a472ae0c56#diff-a62b0d8c3182db370268ec303b89e1505852bdd17e2f8f7879fa29df33422fa0R218

File: packages/connect/src/data/defaultFeeLevels.ts
Comment (memoriesadrift) [reply packages/connect/src/data/defaultFeeLevels.ts:99]: Please change to 5000 * 12 in that case to be accurate to the spec.

The rationale raised for this when it was added was that the functionality in suite only ever needs 1 signature, i.e. the default fee is always 5000. 
Fees for token account funding etc. are not accounted for in the max either, since they can essentially be arbitrarily high, is that okay?

Comment (gabrielKerekes) [reply packages/connect/src/data/defaultFeeLevels.ts:99]: > Fees for token account funding etc. are not accounted for in the max either, since they can essentially be arbitrarily high, is that okay?

I think those shouldn't be counted as fees.

`5000*12` added in https://github.com/trezor/trezor-suite/pull/9958/commits/8e5bb47a6550e723d4fca78b9d7923111d1b679e.

File: packages/connect/src/device/DeviceCommands.ts
Comment (marekrjpolak) [inline packages/connect/src/device/DeviceCommands.ts]: There is no such thing as `tSOL`?

File: packages/connect/src/data/defaultFeeLevels.ts
Comment (marekrjpolak) [reply packages/connect/src/data/defaultFeeLevels.ts:99]: As far as I know, both `minFee` and `maxFee` represent values per unit (same as `feePerUnit`) and both of them are usually used per (v)byte. So I want to ask, do you plan to make changes in Suite so the default unit for Solana is signature rather than byte, and if you do, shouldn't the `maxFee` be same as the `minFee`? If not, it may be necessary to adjust the values to bytes.

File: packages/connect/src/device/DeviceCommands.ts
Comment (gabrielKerekes) [reply packages/connect/src/device/DeviceCommands.ts]: Indeed, there should also be an `|| coinInfo.shortcut === 'DSOL'` here. It somehow got lost during rebasing as it is present in the `feat/solana` branch. Updated in https://github.com/trezor/trezor-suite/pull/9958/commits/e7a2124470db1cb80ea0d6deea222314ab168af5.

File: packages/connect/src/data/defaultFeeLevels.ts
Comment (gabrielKerekes) [reply packages/connect/src/data/defaultFeeLevels.ts:99]: The `minFee` and `maxFee` values are actually not used for Solana as the fee in Suite is constant - there is always only 1 signature per transaction and so the fee is always 5000 Lamports. `feePerUnit` is basically used as `feePerSignature` in `sendFormSolanaActions.ts` ([here](https://github.com/vacuumlabs/trezor-suite/blob/af8edc0616a010b515fb2a59c5882a6dc41fe4d3/packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts#L35)). I changed the values to `-1` in https://github.com/trezor/trezor-suite/pull/9958/commits/a532ce4e2dc77e703779780ac4cd80cc8b59923e and added a comment in https://github.com/trezor/trezor-suite/pull/9958/commits/7778837ab8c6a82d2f3f159441372be643ab42db.

Comment (marekrjpolak) [reply packages/connect/src/data/defaultFeeLevels.ts:99]: Thanks for the info, I'll look at it again in UI PR.

**Review by mroz22 — APPROVED** [review]
Looks ok to me. Lets wait for second approval by @marekrjpolak 

This means that roughly 70% of connect code will be merged to develop. Now our goal is to remove all solana dependencies from Suite layer. So we are probably going to ask you to implement solanaComposeTransaction but lets discuss this over another channel.

**Review by marekrjpolak — APPROVED** [review]

**Comment by gabrielKerekes** [comment]
Ok, can I squash and rebase?

**Comment by gabrielKerekes** [comment]
>  roughly 70% of connect code

What do you mean by this please?

**Comment by gabrielKerekes** [comment]
I'll fix the tests.

**Comment by gabrielKerekes** [comment]
CI should be fixed in https://github.com/trezor/trezor-suite/pull/9958/commits/b18dc71e3b494d86abfbb6230cf595febf913f3c - I expect some of the tests will need to be updated when FW is released. There was also one missing `DSOL` check so I've added that.

The https://github.com/trezor/trezor-suite/pull/9958/commits/a0267eacfee9dab1bab60255eda288c7c7eae11c commit  will be removed, the PR just wasn't being updated after I pushed to our fork's branch, so I tried pushing agian.

**Pull request merged**
